isic/studies/api.py

```diff
@@ -1,29 +1,107 @@
+from isic.core.permissions import SessionAuthStaffUser
+from isic.studies.models import Annotation, Feature, Question, QuestionChoice, Study, StudyTask
 
+annotation_router = Router()
+study_router = Router()
+study_task_router = Router()
 
-class StudyTaskViewSet(viewsets.ReadOnlyModelViewSet):
-    serializer_class = StudyTaskSerializer
-    queryset = StudyTask.objects.all()
-    permission_classes = [IsAdminUser]
 
-    swagger_schema = None
+class AnnotationOut(ModelSchema):
+    class Config:
+        model = Annotation
+        model_fields = ["id", "study", "image", "task", "annotator"]
 
 
-class StudyViewSet(viewsets.ReadOnlyModelViewSet):
-    serializer_class = StudySerializer
-    queryset = Study.objects.all()
-    permission_classes = [IsAdminUser]
+@annotation_router.get(
+    "/", response=list[AnnotationOut], include_in_schema=False, auth=SessionAuthStaffUser
+)
+@paginate(CursorPagination)
+def annotation_list(request: HttpRequest):
+    return Annotation.objects.all()
 
-    swagger_schema = None
 
+@annotation_router.get(
+    "/{id}/", response=AnnotationOut, include_in_schema=False, auth=SessionAuthStaffUser
+)
+def annotation_detail(request: HttpRequest, id: int):
+    return get_object_or_404(Annotation, id=id)
 
-class AnnotationViewSet(viewsets.ReadOnlyModelViewSet):
-    serializer_class = AnnotationSerializer
-    queryset = Annotation.objects.all()
-    permission_classes = [IsAdminUser]
 
-    swagger_schema = None
+class FeatureOut(ModelSchema):
+    class Config:
+        model = Feature
+        model_fields = ["id", "required", "name", "official"]
+
+
+class QuestionChoiceOut(ModelSchema):
+    class Config:
+        model = QuestionChoice
+        model_fields = ["id", "question", "text"]
+
+
+class QuestionOut(ModelSchema):
+    choices: list[QuestionChoiceOut]
+    required: bool = Field(alias="required")
+
+    class Config:
+        model = Question
+        model_fields = ["id", "type", "prompt", "official"]
+
+
+class StudyOut(ModelSchema):
+    class Config:
+        model = Study
+        model_fields = ["id", "created", "creator", "name", "description"]
+
+    features: list[FeatureOut]
+    questions: list[QuestionOut] = Field(alias="questions")
+
+    @staticmethod
+    def resolve_questions(study: Study) -> list[QuestionOut]:
+        # Is there a better way with ninja to add fields from the M2M through-table?
+        vals = []
+        for study_question in study.studyquestion_set.all():
+            study_question.question.required = study_question.required
+            question_out = QuestionOut.from_orm(study_question.question)
+            vals.append(question_out)
+        return vals
+
+
+@study_router.get("/", response=list[StudyOut], include_in_schema=False, auth=SessionAuthStaffUser)
+@paginate(CursorPagination)
+def study_list(request: HttpRequest):
+    return Study.objects.prefetch_related("features", "studyquestion_set__question__choices")
```

Comment:
@danlamanna the previous study-tasks and studies listing endpoints both suffered from n+1 query bugs, both should now be fixed.